### ISSUE TITLE
Improve GUI Design

### ISSUE THREAD

**Issue by jeshuamaxey** [guillefix/augmath#4]
The GUI could be more intuitive, with better placement of buttons and clearer descriptions of what some potentially ambiguous actions will do

**Comment by jeshuamaxey** [comment]
I've begun work on this. Feedback is very welcome. See [this screenshot](http://imgur.com/mbYZrbZ). This work is currently in a branch on my own fork of this repo, [found here](https://github.com/jeshuamaxey/augmath/tree/organising-manipulations).

Things I'm thinking of doing next:
- [ ] improve equation canvas
- [ ] collapsable toolbar sections
- [ ] better styles for toggle-able options (eg multiselect) by making use of the active class from bootstrap

**Comment by guillefix** [comment]
I just saw it, and it looks good! I was thinking that, to keep things organized, the manipulations could be divided into Algebraic and Free editing. The former are only algebraically valid manipulations, while the latter are to give you more flexibility. The current manipulations fit like this:
- Algebraic
  - Change side
  - Divide by factor
  - Root by power
  - Evaluate/Simplify
  - Move selection
  - Add to both sides
  - Split fraction
  - Distribute in
  - Factor Out
- Free editing
  - Remove
  - Unbracket
  - Replace selection with

These could be subclasified further, but I think that'd make it harder to use, at least with the number of manipulations we have at the moment. In any case, when we have more manipulations I think we should have a way of having the most common ones on top, and more accessible, than the least common ones (in the future, I think it'd be cool that people could have their own custom manipulations, by clicking some "Record" button, doing stuff, and saving that as a custom manipulation, they can in the future invoke)

**Comment by jeshuamaxey** [comment]
I was wondering how to categorise them. I like your suggestion above. I don't think they need to be subcategorised further. If more manipulations are added, I think a search led UI would be best, particularly for community contributed manipulations.

I know you raised it in the todo section of the readme, but it'll be cool to have some of the algebraic manipulations unselectable when they don't apply, for example the user shouldn't be able to click "change side" if they've selected a denominator.

**Comment by guillefix** [comment]
Yes I think that'd be a good thing to have (the unselectable manipulations depending on selection). I was also thinking of how useful may it be to have a top toolbar with the most common manipulations appearing as some icons (which we'll have to design or find), leaving the left toolbar for the full name and full list of manipulations.

I was even thinking of making it even easier to manipulate (i.e., less cursor-moving), by making the common manipulations for given selection appear in some bubble above or below the selection, or alternatively, as a custom right-click context menu.

**Comment by jeshuamaxey** [comment]
These are all good ideas. I think it'd be best to sketch each and prioritise them. I might have time this evening or this weekend to work more on the UI

**Comment by guillefix** [comment]
Applied some of the ideas in 502b90a11a320357941bf1b4082ce0a8e816b52b
